packages/node_modules/pouchdb-adapter-indexeddb/src/allDocs.js

```diff
@@ -141,79 +140,115 @@ export default function (txn, metadata, opts, callback) {
     }
   }
 
-  function allDocsInner(doc) {
-    if (doc.error && keys) {
-      // key was not found with "keys" requests
-      results.push(doc);
-      return true;
-    }
-
-    var row = {
-      id: doc.id,
-      key: doc.id,
-      value: {
-        rev: doc.rev
-      }
+  function onTxnComplete() {
+    const returnVal = {
+      total_rows: metadata.doc_count,
+      offset: 0,
+      rows: results
     };
+    /* istanbul ignore if */
+    if (opts.update_seq) {
+      returnVal.update_seq = metadata.seq;
+    }
 
-    var deleted = doc.deleted;
-    if (deleted) {
-      if (keys) {
-        results.push(row);
-        row.value.deleted = true;
-        row.doc = null;
-      }
-    } else if (skip-- <= 0) {
-      results.push(row);
-      if (opts.include_docs) {
-        include_doc(row, doc);
-      }
-      if (--limit === 0) {
-        return false;
-      }
+    if (processing.length) {
+      Promise.all(processing).then(function () {
+        callback(null, returnVal);
+      });
+    } else {
+      callback(null, returnVal);
     }
-    return true;
   }
 
-  function onTxnComplete() {
-    Promise.all(processing).then(function () {
-      var returnVal = {
-        total_rows: metadata.doc_count,
-        offset: 0,
-        rows: results
-      };
+  const dbIndex = docStore.index('_isDeleted_id');
```

Comment:
Can we give this a more descriptive name, like `deletedIdsIndex`?

Reply:
It's a compound index for all docs,  of `deleted` and `id` properties... I think `deletedIds` implies that it's the IDs of deleted docs.

How about `'deleted,id'`?  (not 100% sure this is legal)

Reply:
Ahh ok. I was just talking about the js variable name, not the index name!

Reply:
Oops :grin: 

There's only one index being used in this file; is it helpful to give `dbIndex` a more specific name?

Reply:
It took me about 5 seconds to work out what this actually was (and even then apparently I got it wrong) so yeah, but not critical!

Reply:
Yeah that makes it more obvious to me

Reply:
Done :+1: